pkg/cliplugins/claims/plugin/edit.go

```diff
@@ -0,0 +1,193 @@
+	}
+
+	_, err = fmt.Fprintf(e.Out, "Running interactive prompt. Enter (%s/%s/%s)\n", "Yes", "No", "Skip")
+	if err != nil {
+		return err
+	}
+
+	allErrors := []error{}
+	for _, apibinding := range apibindings {
+		openClaims := getOpenClaims(apibinding.Status.ExportPermissionClaims, apibinding.Spec.PermissionClaims)
+		if len(openClaims) == 0 {
+			_, err = fmt.Fprintf(e.Out, "No open claims found for apibinding: %q\n", apibinding.Name)
+			if err != nil {
+				allErrors = append(allErrors, err)
+			}
+		}
+
+		// List open claims and update `spec.AcceptableClaims` with the status based on the input of the user.
+		for _, openClaim := range openClaims {
+			// print the prompt and infer the user input.
+			action, err := getRequiredInput(e.In, e.Out, apibinding.Name, openClaim.Group, openClaim.Resource)
+			if err != nil {
+				allErrors = append(allErrors, err)
+			}
+
+			// Update apibinding based on the user input.
+			err = updateAPIBinding(ctx, action, openClaim, &apibinding, currentClusterName, kcpClusterClient)
+			if err != nil {
+				allErrors = append(allErrors, err)
+			}
+		}
+	}
+	return utilerrors.NewAggregate(allErrors)
+}
+
+// getOpenClaims gets the claims which are in a set of `exportClaims` but not in `spec.AcceptableClaims` of the apibinding.
+func getOpenClaims(exportClaims []apiv1alpha1.PermissionClaim, aceptableClaims []apiv1alpha1.AcceptablePermissionClaim) []apiv1alpha1.PermissionClaim {
+	openPermissionClaims := []apiv1alpha1.PermissionClaim{}
+
+	// Find the subset which is in exportClaims but not in AcceptableClaims.
+	exportClaimsMap := map[apiv1alpha1.GroupResource]apiv1alpha1.PermissionClaim{}
+	for _, e := range exportClaims {
+		exportClaimsMap[e.GroupResource] = e
+	}
+
+	acceptableClaimsMap := map[apiv1alpha1.GroupResource]apiv1alpha1.PermissionClaim{}
+	for _, a := range aceptableClaims {
+		acceptableClaimsMap[a.GroupResource] = a.PermissionClaim
+	}
+
+	for gr, cl := range exportClaimsMap {
+		if _, ok := acceptableClaimsMap[gr]; !ok {
+			openPermissionClaims = append(openPermissionClaims, cl)
+		}
+	}
+	return openPermissionClaims
+}
+
+// updateAPIBinding send a request to API server to update the APIBinding with the updated status of the claim.
+func updateAPIBinding(ctx context.Context, action ClaimAction, permissionClaim apiv1alpha1.PermissionClaim, apibinding *apiv1alpha1.APIBinding, clusterName logicalcluster.Name, kcpclusterclient kcpclient.ClusterInterface) error {
+	if action == SkipClaim {
+		return nil
+	}
+
+	var state apiv1alpha1.AcceptablePermissionClaimState
+	if action == AcceptClaim {
+		state = apiv1alpha1.ClaimAccepted
+	} else if action == RejectClaim {
+		state = apiv1alpha1.ClaimRejected
+	}
+
+	apibinding.Spec.PermissionClaims = append(apibinding.Spec.PermissionClaims, apiv1alpha1.AcceptablePermissionClaim{PermissionClaim: permissionClaim, State: state})
+
+	_, err := kcpclusterclient.Cluster(clusterName).ApisV1alpha1().APIBindings().Update(ctx, apibinding, metav1.UpdateOptions{})
```

Comment:
You likely want to do the update once, aggregating all changes at once.

Reply:
A single APIBinding is updated once with all the changes in its permission claims. Do you mean updating all the apibindings in a single go? I'm not sure if that would make a difference since each apibinding will require a separate http call.

Reply:
My bad, sorry, please ignore my comment! You were right, it was getting updated for each permission claim, fixed it!